legend-engine-authentication-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/auth/AuthenticationMethod.java

```diff
@@ -0,0 +1,47 @@
+package org.finos.legend.engine.protocol.pure.v1.model.packageableElement.auth;
+
+import org.eclipse.collections.impl.list.mutable.FastList;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.authentication.AuthenticationSpec;
+import org.finos.legend.engine.shared.core.identity.Credential;
+import org.finos.legend.engine.shared.core.identity.Identity;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.List;
+
+public abstract class AuthenticationMethod<Spec, Cred extends Credential>
```

Comment:
Why AuthenticationMethod? I believe its about generating Credential from AuthenticationSpec and Identity maybe we should name it a bit differently to reflect that

Reply:
I was initially looking for a name to distinguish this from AuthenticationStrategy.

But I agree, AuthenticationMethod is misleading .. especially because authentication involves presenting some credential info to a system. In this case, these classes are just creating the credentials and not presenting it to a system.

So maybe CredentialSupplier ? CredentialProvider ?